naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/GraphQLController.scala

```diff
@@ -71,7 +72,29 @@ class GraphQLController @Inject() (
       case _ => None
     }.getOrElse(Json.obj())
 
-    executeQuery(query, request, variables, operation)
+    executeQuery(query, request, variables, operation).map { res =>
+      Ok(res.response)
+    }
+  }
+
+  def graphqlBatch = Action.async(parse.json) { request =>
+
+    val queries = request.body.as[List[JsObject]]
+    val resultsFut = Future.traverse(queries) { queryObj =>
+      val query = (queryObj \ "query").as[String]
+      val operation = (queryObj \ "operationName").asOpt[String]
+
+      val variables = (queryObj \ "variables").toOption.flatMap {
+        case JsString(vars) => Some(parseVariables(vars))
+        case obj: JsObject => Some(obj)
+        case _ => None
+      }.getOrElse(Json.obj())
+
+      executeQuery(query, request, variables, operation)
+    }
+    resultsFut.map { results =>
+      Ok(Json.arr(results.map(_.response)))
```

Comment:
something is odd about what's being returned here. ![image](https://cloud.githubusercontent.com/assets/1118534/20330450/d8d26d82-ab53-11e6-8834-2316a98ea247.png)

unfortunately im not certain what's going on here other than there's too many maps or something here